community/kernel/src/main/java/org/neo4j/unsafe/impl/batchimport/input/csv/CsvInput.java

```diff
@@ -107,11 +108,12 @@ private void checkUniqueCharacter( Map<Character,String> characters, char charac
             @Override
             public InputIterator<InputNode> iterator()
             {
-                return new InputGroupsDeserializer<>( nodeDataFactory.iterator(),
-                        nodeHeaderFactory, config, idType, maxProcessors, (dataStream, dataHeader, decorator) ->
+                DeserializerFactory<InputNode> factory = (dataStream, dataHeader, decorator, validator) ->
```

Comment:
Talking about readability. 
Code reads better now with splitting initialisation of `DeserializerFactory` from `InputGroupsDeserializer` constructor call.
It still reads pretty bad though. Can we split it up even more? Or maybe extract some initialisations into helper methods?